From 8fe380fb1e57338c6239381154d72d495e644818 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Wed, 29 Mar 2017 13:26:26 +0800
Subject: [PATCH 1622/5242] MLK-14541 dmaengine: fsl-edma: remove the
 duplicated code

commit  cb2085287582c0a8904cae2e01095f13e1de3177 from
https://source.codeaurora.org/external/imx/linux-imx.git

The function .fsl_edma_irq_init() has been called twice in .probe(), which
cause all dma controller registered failed.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/fsl-edma.c |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/dma/fsl-edma.c b/drivers/dma/fsl-edma.c
index 795f98f..0951bce 100644
--- a/drivers/dma/fsl-edma.c
+++ b/drivers/dma/fsl-edma.c
@@ -1079,6 +1079,7 @@ static int fsl_edma_probe(struct platform_device *pdev)
 
 	}
 
+	edma_writel(fsl_edma, ~0, fsl_edma->membase + EDMA_INTR);
 	if (fsl_edma->version == 1)
 		ret = fsl_edma_irq_init(pdev, fsl_edma);
 	else
@@ -1105,11 +1106,6 @@ static int fsl_edma_probe(struct platform_device *pdev)
 		fsl_chan->vchan.chan.chan_id = 0;
 	}
 
-	edma_writel(fsl_edma, ~0, fsl_edma->membase + EDMA_INTR);
-	ret = fsl_edma_irq_init(pdev, fsl_edma);
-	if (ret)
-		return ret;
-
 	dma_cap_set(DMA_PRIVATE, fsl_edma->dma_dev.cap_mask);
 	dma_cap_set(DMA_SLAVE, fsl_edma->dma_dev.cap_mask);
 	dma_cap_set(DMA_CYCLIC, fsl_edma->dma_dev.cap_mask);
-- 
1.7.9.5

